webapp/src/modules/parcels/sagas.js

```diff
@@ -37,6 +46,57 @@ function* handleParcelsRequest(action) {
   }
 }
 
+function* handleParcelRequest(action) {
+  try {
+    const { x, y } = action
+    const nw = buildCoordinate(x, y)
+    const se = buildCoordinate(x, y)
+
+    let [parcels, data] = yield all([
+      api.fetchParcels(nw, se),
+      api.fetchParcelData(x, y)
+    ])
+
+    parcels = parcels.map(parcel => Object.assign(parcel, { data }))
+
+    yield put({
+      type: FETCH_PARCEL_SUCCESS,
+      parcels
+    })
+  } catch (error) {
+    console.warn(error)
+    yield put({
+      type: FETCH_PARCEL_FAILURE,
+      error: error.message
+    })
+  }
+}
+
+function* handleParcelDataRequest(action) {
+  try {
+    const { x, y } = action
+    if (!inBounds(x, y)) return
+
+    const parcels = yield select(getParcels)
+    const parcel = parcels[buildCoordinate(x, y)]
+    if (!parcel) return
```

Comment:
Question for ` if (!inBounds(x, y)) return` and `if (!parcel) return`. Even though `handleHoverParcel` smartly avoids calls to this method when it's not necessary, shouldn't we `throw` here anyways?